libs/forecasting/src/lib/ForecastingDashboard/Controls/TransformationCreationDialog.tsx

```diff
@@ -0,0 +1,242 @@
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public constructor(props: ITransformationCreationDialogProps) {
+    super(props);
+    this.state = { transformationValue: 2 };
+  }
+
+  public componentDidUpdate(
+    prevProps: Readonly<ITransformationCreationDialogProps>
+  ): void {
+    if (this.props.isVisible !== prevProps.isVisible) {
+      this.setState({
+        transformationFeature: undefined,
+        transformationName: undefined,
+        transformationOperation: undefined,
+        transformationValue: 2
+      });
+    }
+  }
+
+  public render(): React.ReactNode {
+    const classNames = forecastingDashboardStyles();
+
+    let transformationNameErrorMessage = undefined;
+    if (!this.state.transformationName) {
+      transformationNameErrorMessage =
+        localization.Forecasting.TransformationCreation
+          .scenarioNamingInstructions;
+    } else if (
+      this.state.transformationName &&
+      this.props.transformations.has(this.state.transformationName)
+    ) {
+      transformationNameErrorMessage =
+        localization.Forecasting.TransformationCreation
+          .scenarioNamingCollisionMessage;
+    }
+
+    let transformationValueErrorMessage = undefined;
+    if (
+      this.state.transformationOperation &&
+      (this.state.transformationValue <
+        this.state.transformationOperation.minValue ||
+        this.state.transformationValue >
+          this.state.transformationOperation.maxValue ||
+        this.state.transformationOperation.excludedValues.includes(
+          this.state.transformationValue
+        ))
+    ) {
+      transformationValueErrorMessage = localization.formatString(
+        localization.Forecasting.TransformationCreation.valueErrorMessage,
+        this.state.transformationOperation.displayName,
+        this.state.transformationOperation.minValue,
+        this.state.transformationOperation.maxValue,
+        this.state.transformationOperation.excludedValues.toString()
+      );
+    }
+
+    let transformationCombinationErrorMessage = undefined;
+    if (
+      this.state.transformationOperation &&
+      this.state.transformationFeature
+    ) {
+      // ensure the current selection isn't a duplicate
+      const transformation = this.createTransformation();
+      if (transformation) {
+        this.props.transformations.forEach((existingTransformation) => {
+          const equalsResult = transformation.equals(existingTransformation);
+          if (equalsResult) {
+            transformationCombinationErrorMessage =
+              localization.Forecasting.TransformationCreation
+                .invalidCombinationErrorMessage;
+          }
+        });
+      }
+    }
+    return (
+      <Dialog
+        hidden={!this.props.isVisible}
+        dialogContentProps={{
+          closeButtonAriaLabel: "Close",
```

Comment:
localization for aria label?

Reply:
Oh, I had not heard of localization for aria labels. After a quick search I found examples for both localization and lack thereof. I will not address it in this PR, but I'll tackle all aria labels in a separate PR. I hope that's fine by you.